kctf-operator/pkg/controller/challenge/challenge_check.go

```diff
@@ -0,0 +1,84 @@
+// File that checks if all configurations are correctly set
+package challenge
+
+import (
+	kctfv1alpha1 "github.com/google/kctf/pkg/apis/kctf/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+func CheckDeployed(challenge *kctfv1alpha1.Challenge, deployment *appsv1.Deployment) bool {
+	// First, check if the challenge is ready and, if not, set replicas to 0
+	if challenge.Spec.Deployed == false && *deployment.Spec.Replicas != 0 {
+		var numReplicas int32 = 0
```

Comment:
Yes, we could move to it, I just created other file because maybe the code would be too big in challenge-controller and, then, it would be harder for someone that wants to make changes in this code or simply understand it.

Maybe we could also change the world "check", so it's clearer to who reads the code.

Reply:
changing the name sgtm :)

Reply:
Yeah, I think the problem here is with naming.
So basically what happens here is these functions check the pushed config ("desired state") and actual objects ("actual state") and make sure that actual state matches the desired state.

maybe smth like `ensureDeployed`? (or other better name)

btw, does the function have to be exported? They are all in the same package, right?

Reply:
+1, changing the name would resolve my confusion

Reply:
Ok, so I will replace all "check" for "ensure" if it's ok.

Yes, I left all these five files in the same package and, then, the functions from these files don't need to be exported since they are from the same package.

Reply:
Actually what do you think about "verify"? It sounds better for me.

Reply:
verify has the same problem as check: it assumes that you only checking/verifying values using some sort of validation.

I think smth like apply/update/ensure are better verbs for these functions